### PR TITLE
cv: image opti

### PR DESCRIPTION
**Pull request by anthonycorletti**
* [ ] I have added tests to `tests` to cover my changes.
* [ ] I have updated `docs/`, if necessary.
* [ ] I have updated the `README.md`, if necessary.

***What existing issue does this pull request close?***

No issue. This uploads images to minio and retrieves them via presigned url.

***How are these changes tested?***

Locally!

***Demonstration***

Screenshot 2023-02-14 at 10 07 15 PM

### CODE REVIEW

**Comment by anthonycorletti** [comment]
the text field will be `project_id/image_id.image_extension` we still need project_id for authorization.

File: dataquality/clients/api.py [hunk 1]
```diff
@@ -560,19 +560,35 @@ def wait_for_run(
...
+        params = {
+            "api_url": config.api_url,
+            "bucket_name": bucket_name,
+            "object_name": object_name,
+            "method": method.upper(),
+        }
+        if image_id is not None:
+            params["image_id"] = image_id
+        elif project_id is not None:
+            params["project_id"] = project_id
```
Comment (Ben-Epstein) [inline dataquality/clients/api.py:579]: can they ever both be not-non?

File: dataquality/clients/objectstore.py [hunk 2]
```diff
@@ -11,26 +12,31 @@
 from dataquality.core.auth import api_client
 from dataquality.utils.file import get_file_extension
 
+IMAGES_BUCKET_NAME = "galileo-images"
```
Comment (Ben-Epstein) [inline dataquality/clients/objectstore.py:15]: the api makes this configurable with environment variables (along with the others). SHould we consider making these configurable too? Probably an edge case, but the API PR made me think about it

File: dataquality/utils/cv.py [hunk 3]
```diff
@@ -46,3 +54,61 @@ def _bytes_to_b64_str(img_bytes: bytes, img_path: Optional[str] = None) -> str:
...
+    image_id: Optional[UUID4] = None,
+) -> str:
+    if image_id is None:
+        image_id = uuid4()
+
+    img_bytes = BytesIO()
+    if img_path is not None:
+        with open(img_path, "rb") as f:
+            img_bytes.write(f.read())
+        _format = img_path.split(".")[-1].upper()
```
Comment (Ben-Epstein) [inline dataquality/utils/cv.py:71]: why `.upper` here and then `.lower` on line 82?

File: dataquality/utils/cv.py [hunk 4]
```diff
@@ -1,10 +1,18 @@
...
 
 from PIL import Image
+from pydantic import UUID4
 
+from dataquality import config
+from dataquality.clients.objectstore import ObjectStore
+from dataquality.exceptions import GalileoException
+
+object_store = ObjectStore()
 B64_CONTENT_TYPE_DELIMITER = ";base64,"
```
Comment (Ben-Epstein) [inline dataquality/utils/cv.py:16]: is this used anymore? Maybe it can be removed

File: dataquality/clients/api.py [hunk 5]
```diff
@@ -560,19 +560,35 @@ def wait_for_run(
                 )
 
     def get_presigned_url(
-        self, project_id: str, method: str, bucket_name: str, object_name: str
+        self,
+        method: str,
+        bucket_name: str,
+        object_name: str,
+        project_id: Optional[str] = None,
+        image_id: Optional[str] = None,
```
Comment (Ben-Epstein) [inline dataquality/clients/api.py:568]: hm It doesn't look like you're using this `image_id` anywhere when calling `get_presigned_url` - was that potentially an old idea you moved away from?

File: dataquality/clients/objectstore.py [hunk 6]
```diff
@@ -11,26 +12,31 @@
...
         progress: bool = True,
+        bucket_name: Optional[str] = None,
     ) -> None:
+        if not bucket_name:
+            bucket_name = self.ROOT_BUCKET_NAME
         url = api_client.get_presigned_url(
             project_id=object_name.split("/")[0],
             method="put",
-            bucket_name=self.ROOT_BUCKET_NAME,
+            bucket_name=bucket_name,
```
Comment (Ben-Epstein) [inline dataquality/clients/objectstore.py:39]: could be fancy like and remove the top `if` 😎 
```suggestion
            bucket_name=bucket_name or self.ROOT_BUCKET_NAME,
```

File: dataquality/clients/api.py (hunk 1 above)
Comment (anthonycorletti) [reply dataquality/clients/api.py:579]: ah! this is a bit crufty. cleaning it up.

File: dataquality/clients/objectstore.py (hunk 2 above)
Comment (anthonycorletti) [reply dataquality/clients/objectstore.py:15]: great idea! made a ticket for it https://app.shortcut.com/galileo/story/3801

File: dataquality/utils/cv.py (hunk 3 above)
Comment (anthonycorletti) [reply dataquality/utils/cv.py:71]: good question! i dont think it has to be this way. i did the upper to keep the format returned by the PIL consistent when it was passed to .save. and then lower for the file extension.

File: dataquality/clients/api.py (hunk 5 above)
Comment (anthonycorletti) [reply dataquality/clients/api.py:568]: yes!

File: dataquality/clients/objectstore.py (hunk 6 above)
Comment (anthonycorletti) [reply dataquality/clients/objectstore.py:39]: pinkies up!

**Review by Ben-Epstein — APPROVED** [review]

**Pull request merged**